src/passes/DeadCodeElimination.cpp

```diff
@@ -110,11 +115,6 @@ struct DeadCodeElimination
             replaceCurrent(remainingChildren[0]);
           } else {
             replaceCurrent(builder.makeBlock(remainingChildren));
```

Comment:
The only danger with a Pop is the new block made here, so I suggest that we set a boolean `addedBlock` to true in this location. And we can rename `needEHFixups` to `hasPop`. Then below, the check `if (needEHFixups)` can be `if (hasPop && addedBlock)`, which will be a little more precise.

Reply:
Good idea. Done.